api/resolvers/user.js

```diff
@@ -1082,19 +1083,37 @@ export default {
 
       return user.streak
     },
-    gunStreak: async (user, args, { models }) => {
+    hasSendWallet: async (user, args, { models }) => {
       if (user.hideCowboyHat) {
-        return null
+        return false
       }
 
-      return user.gunStreak
+      const vaultEntry = await models.vaultEntry.findFirst({
+        where: {
+          userId: user.id,
+          wallet: {
+            enabled: true
+          }
+        }
+      })
+      return !!vaultEntry
     },
-    horseStreak: async (user, args, { models }) => {
+    hasRecvWallet: async (user, args, { models }) => {
       if (user.hideCowboyHat) {
-        return null
+        return false
       }
 
-      return user.horseStreak
+      const wallets = await models.wallet.findMany({
+        where: {
+          userId: user.id,
+          enabled: true
+        }
+      })
+
+      return wallets.some(({ type, wallet: config }) => {
+        const def = getWalletByType(type)
+        return canReceive({ def, config })
+      })
```

Comment:
This adds two new db queries to each page load for logged in users? did you consider mitigating this?

The db columns suggested to me that you were going to update these when wallets were set and unset ... but the resolvers don't use them.

Reply:
You are right, I should use the columns, see https://github.com/stackernews/stacker.news/pull/2040#discussion_r2025612148